service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/update/UpdateControlledBigQueryDatasetsLifetimeStep.java

```diff
@@ -0,0 +1,112 @@
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.http.HttpStatus;
+
+public class UpdateControlledBigQueryDatasetsLifetimeStep implements Step {
+  private final ResourceDao resourceDao;
+
+  public UpdateControlledBigQueryDatasetsLifetimeStep(ResourceDao resourceDao) {
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var workingMap = context.getWorkingMap();
+    validateRequiredEntries(
+        workingMap,
+        CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP,
+        CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_PARTITION_LIFETIME_MAP,
+        CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP);
+
+    System.out.println("WORKING MAP: " + workingMap);
+
+    Map<UUID, Long> resourceIdToDefaultTableLifetimeMap =
+        workingMap.get(
+            CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP,
+            new TypeReference<HashMap<UUID, Long>>() {});
+    Map<UUID, Long> resourceIdToDefaultPartitionLifetimeMap =
+        workingMap.get(
+            CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_PARTITION_LIFETIME_MAP,
+            new TypeReference<HashMap<UUID, Long>>() {});
+
+    Map<UUID, String> resourceIdsToWorkspaceIdMap =
+        workingMap.get(CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP, new TypeReference<>() {});
+    List<ControlledResource> updatedResources = new ArrayList<>();
+
+    List<ControlledResource> controlledBigQueryDatasets =
+        context
+            .getWorkingMap()
+            .get(CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME, new TypeReference<>() {});
+
+    System.out.println("LIST OF BQ datasets " + controlledBigQueryDatasets);
+
+    System.out.println("id to part lifetime " + resourceIdToDefaultPartitionLifetimeMap);
+
+    System.out.println("id to table lifetime " + resourceIdToDefaultTableLifetimeMap);
+
+    for (var resource : controlledBigQueryDatasets) {
+      var id = resource.getResourceId();
+      System.out.println("BQ dataset without lifetime " + id);
+      boolean updated =
+          resourceDao.updateBigQueryDatasetDefaultTableLifetime(
+                  resource.castByEnum(CONTROLLED_GCP_BIG_QUERY_DATASET),
+                  resourceIdToDefaultTableLifetimeMap.get(id))
```

Comment:
should you only call updateBqDatasetDefaultTableLifetime only when the map has the key?